Controller/Redirect/Process.php

```diff
@@ -62,14 +68,21 @@ protected function redirectSuccess()
 
         //add clientid - GA tracking
         $clientId = $this->getRequest()->getParam('clientId');
+
+
+        $queryArguments = [];
+        parse_str(parse_url($url, PHP_URL_QUERY), $queryArguments);
```

Comment:
would be good to check if there aren't already some functions/classes that wrap around these functions... (part of magento core/framework).